Monika After Story/game/script-topics.rpy

```diff
@@ -663,11 +735,32 @@ label mas_rerandom:
 
     else:
         show monika at t11
-        $ mas_showEVL(topic_choice, mas_bookmarks_derand.ev_db_code, _random=True)
-        #Pop the derandom
-        $ mas_bookmarks_derand.persist_var.pop(mas_bookmarks_derand.persist_var.index(topic_choice))
-        #Prep the renpy substitution
-        $ talk_about_more_text = renpy.substitute(mas_bookmarks_derand.talk_about_more_text)
+        python:
+            rerand_ev = mas_getEV(topic_choice)
+
+            if rerand_ev:
+                #Rerandom the ev
+                rerand_ev.random = True
+
+                #Run the rerandom callback function
+                rerandom_callback_data = rerand_ev.rules.get("rerandom_callback", None)
+                if isinstance(rerandom_callback_data, dict) and callable(rerandom_callback_data["function"]):
```

Comment:
I would just do a `rerandom_callback` None check here so we can get logging info if the rules dict gets corrupted somehow (aka is not set to a dict, or function is not set, etc...)
```suggestion
                if rerandom_callback_data is not None:
```